File: BotSdk/botclient/src/main/java/com/kore/data/repository/jwt/JwtRepository.kt
package com.kore.data.repository.jwt

import com.kore.common.Result
import com.kore.network.api.response.BotAuthorizationResponse
import com.kore.network.api.response.JwtTokenResponse
import com.kore.network.api.response.RtmUrlResponse

interface JwtRepository {
    fun generateJwt(identity: String, clientSecret: String, clientId: String): String
    fun generateJwtForAPI(identity: String, clientSecret: String, clientId: String): String
    suspend fun getJwtToken(url: String, clientId: String, clientSecret: String, identity: String): Result<JwtTokenResponse?>
    suspend fun getJwtGrant(jwtToken: HashMap<String, Any>): Result<BotAuthorizationResponse?>
    suspend fun getRtmUrl(token: String, optParameterBotInfo: HashMap<String, Any>): Result<RtmUrlResponse?>
    suspend fun getRtmUrl(token: String, botInfo: HashMap<String, Any>, isReconnect: Boolean): Result<RtmUrlResponse?>
}

File: KoreAiChatApp/src/services/JwtRepository.ts
import {
  JwtTokenResponse,
  BotAuthorizationResponse,
  RtmUrlResponse
} from '../types/BotConfigModel';

export class JwtRepository {
  
  public async getJwtToken(
    jwtServerUrl: string,
    clientId: string,
    clientSecret: string,
    identity: string
  ): Promise<JwtTokenResponse | null> {
    try {
      const response = await fetch(jwtServerUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          clientId,
          clientSecret,
          identity
        })
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      const data = await response.json();
      return data;
    } catch (error) {
      console.error('Error getting JWT token:', error);
      throw error;
    }
  }

  public async getJwtGrant(request: Record<string, any>): Promise<BotAuthorizationResponse | null> {
    try {
      // This would be the actual Kore.ai API endpoint for JWT grant
      const response = await fetch('https://platform.kore.ai/api/1.1/oAuth/token', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(request)
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      const data = await response.json();
      return data;
    } catch (error) {
      console.error('Error getting JWT grant:', error);
      throw error;
    }
  }

  public async getRtmUrl(
    accessToken: string,
    request: Record<string, any>
  ): Promise<RtmUrlResponse | null> {
    try {
      // This would be the actual Kore.ai API endpoint for RTM URL
      const response = await fetch('https://platform.kore.ai/api/1.1/rtm/start', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${accessToken}`
        },
        body: JSON.stringify(request)
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      const data = await response.json();
      return data;
    } catch (error) {
      console.error('Error getting RTM URL:', error);
      throw error;
    }
  }
}